app.js

```diff
@@ -0,0 +1,196 @@
+const express = require('express')
+const environment = process.env.NODE_ENV || 'development'
+const configuration = require('./knexfile')[environment]
+const database = require('knex')(configuration)
+
+const app = express()
+app.use(express.json())
+app.get('/', (request, response) => {
+  response.send('Hello World')
+});
+
+// Palette Endpoints
+app.get('/api/v1/palettes', (request, response) => {
+  database('palettes').select()
+    .then(palettes => {
+      response.status(200).json(palettes)
+    })
+    .catch(error => {
+      response.status(500).json({
+        error
+      })
+    })
+});
+
+app.get('/api/v1/palettes/:id', (request, response) => {
+  const reqId = request.params.id
+  database('palettes').where('palette_id', reqId).select()
+    .then(palette => {
+      if (palette.length) {
+        response.status(200).json(palette)
+      } else {
+        response.status(404).json(`Palette with id ${reqId} was not found`)
+      }
+    })
+    .catch(error => {
+      response.status(500).json({
+        error
+      })
+    })
+});
+
+app.post('/api/v1/palettes', (request, response) => {
+  const newPalette = request.body
+
+  for(let requiredParameters of ['palette_name', 'project_id', 'color_1', 'color_2', 'color_3', 'color_4', 'color_5']) {
+    if (newPalette[requiredParameters] === undefined) {
+      response.status(422).json({ error: `Expected format: { palette_name: <String>}, project_id: <Integer>, color_1: <String>, color_2: <String>, color_3: <String>, color_4: <String>, color_5: <String>. You are missing a "${requiredParameters}"`})
+    } else {
+      database('palettes').insert(newPalette, 'palette_id')
+        .then(palette => {
+          response.status(201).json(palette)
+        })
+        .catch(error => {
+          response.status(500).json({
+            error
+        })
+      })
+    }
+  }
+});
+
+app.put('/api/palettes/:id', (request, response) => {
+  const updatedPalette = request.body
+  // okay no required params? or should we require name, just doesn't NEED to be updated?
```

Comment:
I'm thinking on the front-end we can disable the 'submit' for an edit unless they make a change and then this method will never run unless they make a change. I think having the default values in this method is great!!

Reply:
Sounds good!